airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -104,101 +118,249 @@ public AirbyteConnectionStatus check(JsonNode config) {
     }
   }
 
-  protected String getCurrentDatabaseName(DSLContext ctx) {
-    return ctx.select(currentSchema()).fetch().get(0).getValue(0, String.class);
-  }
-
   @Override
   public AirbyteCatalog discover(JsonNode config) throws Exception {
-    try (final Database database = createDatabase(config)) {
+    try (final JdbcDatabase database = createDatabase(config)) {
       return new AirbyteCatalog()
-          .withStreams(getTables(database)
-              .stream()
-              .map(t -> CatalogHelpers.createAirbyteStream(t.getName(), t.getFields()))
-              .collect(Collectors.toList()));
+          .withStreams(getTables(
+              database,
+              Optional.ofNullable(config.get("database")).map(JsonNode::asText),
```

Comment:
We should avoid using optionals in parameters. They should only be used for return types.

As-is they are simply not providing any value, they are just making the code in `getTable` more complex.

Reply:
Should we commit to using `@Nullable` or `@NotNullable` instead?

Reply:
I don't believe the decision is one or the other. 

Using Optional for parameters is abusing what they were created for. They ultimately achieve the exact same thing as passing null.

Using `@Nullable` & `@NotNullable` is just a clutch and a bad one because it only works well when all the code and deps are properly annotated. othwerwise it gives you a false sense of confidence.

What is wrong with having `Preconditions.check...` to ensure that it is not null. or to have a `if null`?

Reply:
(fun that there is a github account called `@nullable`)

Reply:
@michel-tricot decided not to block this PR on this. Separate PR here to keep talking about it. WAnt to see if I understand what you're thinking. https://github.com/airbytehq/airbyte/pull/1399